Codeinwp/themeisle-sdk
# Feat/otter page

### Summary
Added a new product page for Otter.
This can be added to any product by using the same filter for the About page:
```php
add_filter( 'sample_plugin_about_us_metadata', function () {
	return [
                 // Top-level page in the dashboard sidebar
		'location'         => 'themes.php',
                 // Logo to display on the page
		'logo'             => 'https://placehold.co/200x50.jpg',
                 // Menu displayed at the top of the page - optional
		'page_menu'        => [
			[ 'text' => 'SDK GitHub Issues', 'url' => esc_url( 'https://github.com/codeinwp/themeisle-sdk/issues' ) ],
			[ 'text' => 'Themeisle', 'url' => esc_url( 'https://themeisle.com' ) ]
		],
		// Condition to show or hide the upgrade menu in the sidebar
		'has_upgrade_menu' => defined( 'SOME_CONST' ),
               // Add predefined product pages to the about page.
		'product_pages'    => [ 'otter-page' ],
		// Upgrade menu item link & text
		'upgrade_link'     => esc_url( 'https://themeisle.com/themes/neve/pricing/' ),
		'upgrade_text'     => 'Upgrade to Pro',
	];
} );
```
See the `product_pages` attribute, which can be conditionally populated with values.
Changed the filter to support the new attribute `product_pages`

The page will be displayed as a tab of the About us page, allowing the user to install Otter.

### Screenshots
<details open>
    <summary>New. tab menu when filter is used (Pic 1)</summary>

![image](https://github.com/Codeinwp/themeisle-sdk/assets/23024731/18e74586-d85b-4317-92d7-b50da0ae2a6f)
</details>

<details>
    <summary>Otter Page (Pic 2)</summary>

![image](https://github.com/Codeinwp/themeisle-sdk/assets/23024731/a1f9411c-7efe-4a8a-b210-7e4a83ebd4fc)
</details>

### Test instructions
1. Use the build of Neve from here: https://github.com/Codeinwp/neve/pull/4031 to test
2. Make sure that you don't have Otter installed
3. Go to the About page from Neve
4. Check that the new Product page for Otter is added as a tab. (see Pic 1)
5. Check the Otter Product page (see Pic 2)
6. Check that it will install the plugin when clicking on the CTA buttons from the product page
7. Check that the Product page is not displayed anymore if Otter is Installed (and/or not active)

Closes Codeinwp/neve-pro-addon#2519.
